File: Learn/30-Days-Of-Code/Day 29/bitand.py
def max_value(n, k):
    if ((k - 1) | k) <= n:
        return k - 1
    else:
        return k - 2

t = int(input().strip())
for _ in range(t):
    n, k = input().strip().split(' ')
    n, k = [int(n), int(k)]
    print(max_value(n, k))

File: Learn/30-Days-Of-Code/Day 29/bitand.cpp
#include <iostream>

using namespace std;

int maxValue(int n, int k) {
    return ((k - 1) | k) <= n ? k - 1 : k - 2;
}

int main() {
    int t, n, k;
    cin >> t;
    while (t-- > 0) {
        cin >> n >> k;
        cout << maxValue(n, k) << endl;
    }
    return 0;
}
